javaslang/src/main/java/javaslang/collection/TreeMap.java

```diff
@@ -492,6 +492,19 @@ public boolean isEmpty() {
     }
 
     @Override
+    public <K2> TreeMap<K2, V> mapKeys(Function<? super K, ? extends K2> keyMapper) {
+        Objects.requireNonNull(keyMapper, "keyMapper is null");
+        return map((k, v) -> Tuple.of(keyMapper.apply(k), v));
+    }
+
+    @Override
+    public <K2> TreeMap<K2, V> mapKeys(Function<? super K, ? extends K2> keyMapper, Function2<V, V, V> valueMerge) {
+        //todo Hack
+        Comparator<K2> comparator = Comparators.naturalComparator();
```

Comment:
I don't known how to remove this hack.
If I describe `K2` in `TreeMap` as `K2 extends Comparable<? super K2>` then I must describe it with same limit in `SortedMap` (it's ok) and `Map` (this is not ok).